blocks/quilted-image-block/features/quilted-image/default.jsx

```diff
@@ -27,41 +29,164 @@ function QuiltedImage({ customFields }) {
 		fullWidthImage,
 		image1URL,
 		image1AspectRatio,
+		image1Auth,
+		image1Id,
+		image1Alt,
 		overlay1Text,
 		overlay1TextVariant,
 		button1Text,
 		item1Action,
 		button1Variant,
 		image2URL,
 		image2AspectRatio,
+		image2Auth,
+		image2Id,
+		image2Alt,
 		overlay2Text,
 		overlay2TextVariant,
 		button2Text,
 		item2Action,
 		button2Variant,
 		image3URL,
 		image3AspectRatio,
+		image3Auth,
+		image3Id,
+		image3Alt,
 		overlay3Text,
 		overlay3TextVariant,
 		button3Text,
 		item3Action,
 		button3Variant,
 	} = customFields;
 
+	const { isAdmin } = useFusionContext();
+	const { searchableField } = useEditableContent();
+
+	const sharedImageParams = {
+		resizedOptions: {
+			smart: true,
+		},
+		responsiveImages: [200, 400, 600, 800, 1000],
+		width: 600,
+	};
+
+	const image1AuthToken = useContent(
+		!image1Auth && image1Id
+			? {
+					source: "signing-service",
+					query: { id: image1Id },
+			  }
+			: {}
+	);
+
+	const image1AuthTokenObj = {};
+	if (image1AuthToken?.hash) {
+		image1AuthTokenObj[RESIZER_APP_VERSION] = image1AuthToken.hash;
+	}
+
+	const image1Params =
+		image1Id && image1URL
+			? {
+					ansImage: {
+						_id: image1Id,
+						url: image1URL,
+						auth: image1Auth ? JSON.parse(image1Auth) : image1AuthTokenObj,
+					},
+					alt: image1Alt,
+					aspectRatio: ASPECT_RATIO_MAP[image1AspectRatio],
+					...sharedImageParams,
+			  }
+			: {
+					src: image1URL,
+					alt: overlay1Text,
+			  };
+
+	const image2AuthToken = useContent(
+		!image2Auth && image2Id
+			? {
+					source: "signing-service",
+					query: { id: image2Id },
+			  }
+			: {}
+	);
+
+	const image2AuthTokenObj = {};
+	if (image2AuthToken?.hash) {
+		image2AuthTokenObj[RESIZER_APP_VERSION] = image2AuthToken.hash;
+	}
+
+	const image2Params =
+		image2Id && image2URL
+			? {
+					ansImage: {
+						_id: image2Id,
+						url: image2URL,
+						auth: image2Auth ? JSON.parse(image2Auth) : image2AuthTokenObj,
+					},
+					alt: image2Alt,
+					aspectRatio: ASPECT_RATIO_MAP[image2AspectRatio],
+					...sharedImageParams,
+			  }
+			: {
+					src: image2URL,
+					alt: overlay2Text,
+			  };
+
+	const image3AuthToken = useContent(
+		!image3Auth && image3Id
+			? {
+					source: "signing-service",
+					query: { id: image3Id },
+			  }
+			: {}
+	);
+
+	const image3AuthTokenObj = {};
+	if (image3AuthToken?.hash) {
+		image3AuthTokenObj[RESIZER_APP_VERSION] = image3AuthToken.hash;
+	}
+
+	const image3Params =
+		image3Id && image3URL
+			? {
+					ansImage: {
+						_id: image3Id,
+						url: image3URL,
+						auth: image3Auth ? JSON.parse(image3Auth) : image3AuthTokenObj,
+					},
+					alt: image3Alt,
+					aspectRatio: ASPECT_RATIO_MAP[image3AspectRatio],
+					...sharedImageParams,
+			  }
+			: {
+					src: image3URL,
+					alt: overlay3Text,
+			  };
+
```

Comment:
Can we make this a subcomponent that takes in the parameters it needs instead of copy/pasted?

Then you can simplify the additions below and we don't risk someone passing in a stray variable or changing one in a way that makes these different in a cryptic way....